src/averageAges.js

```diff
@@ -14,14 +14,23 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+const MALE = 'f';
+const FEMALE = 'f';
+const ONE_CENTURY = 1000;
+
+function menAverageAge(people, century) {
+  const men = people.filter(person => person.sex === MALE && (century
+    ? Math.ceil(person.died / ONE_CENTURY) === century
+    : true));
+
+  const menTotalAge = men.reduce((acc, man) => {
+    const age = man.died - man.born;
+
+    return acc + age;
+  }, 0);
 
+  return menTotalAge / men.length;
```

Comment:
The function name `menAverageAge` does not follow the naming convention. It should be named `calculateMenAverageAge` to clearly indicate its purpose. 
  Consider checking the division by zero when `men.length` is 0 to avoid NaN result.